### PR TITLE
first recipe for r-arrapply

### PR DESCRIPTION
**Pull request by sgsokol**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by sgsokol** [comment]
@conda-forge/help-r please review

**Comment by dbast** [comment]
@sgsokol Thanks for bringing this.

Can you please update your conda-build version and update the recipe by re-creating it with https://github.com/bgruening/conda_r_skeleton_helper ... no manual edit of the helper output should be required.

Please also keep the group `conda-forge/r` as part of the `recipe-maintainers`. The help updating and maintaining the conda-forge r recipes.

**Comment by sgsokol** [comment]
@dbast Thanks for your help

>  ... no manual edit of the helper output should be required.

Well, I had to manually add "- libopenblas" to "host:". Without that, the checks were failing.

Another point, in my humble opinion there is a logical error in generated build.sh. The beginig of the script looks like 
```
#!/bin/bash
set -o errexit -o pipefail
if [[ ${target_platform} =~ linux.* ]] || [[ ${target_platform} == win-32 ]] || [[ ${target_platform} == win-64 ]] || [[ ${target_platform} == osx-64 ]]; then
  export DISABLE_AUTOBREW=1
  ${R} CMD INSTALL --build .
else
  mkdir -p "${PREFIX}"/lib/R/library/arrApply
  mv ./* "${PREFIX}"/lib/R/library/arrApply
  if [[ ${target_platform} == osx-64 ]]; then
...
```
while I think the intention was:
```
#!/bin/bash
set -o errexit -o pipefail
if [[ ${target_platform} =~ linux.* ]] || [[ ${target_platform} == win-32 ]] || [[ ${target_platform} == win-64 ]] || [[ ${target_platform} == osx-64 ]]; then
  export DISABLE_AUTOBREW=1
  ${R} CMD INSTALL --build .
else
  mkdir -p "${PREFIX}"/lib/R/library/arrApply
  mv ./* "${PREFIX}"/lib/R/library/arrApply
fi
if [[ ${target_platform} == osx-64 ]]; then
...
```
Otherwise, the second if (`if [[ ${target_platform} == osx-64 ]]; then`) will never be evaluated on osx-64 because the first test will be set to TRUE and the control will never go inside the `else` close and thus never reach the second if.

Hoping this helps.

File: recipes/r-arrapply/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    - {{posix}}filesystem        # [win]
+    - {{posix}}make
+    - {{posix}}sed               # [win]
+    - {{posix}}coreutils         # [win]
+    - {{posix}}zip               # [win]
+  host:
+    - r-base
+    - r-rcpp >=0.12.0
+    - r-rcpparmadillo
+    - libopenblas
```
Comment (isuruf) [inline recipes/r-arrapply/meta.yaml]: Please read https://conda-forge.org/docs/maintainer/knowledge_base.html#blas on how to specify blas correctly.

Comment (sgsokol) [reply recipes/r-arrapply/meta.yaml]: Thanks @isuruf for precise doc pointer

**Comment by dbast** [comment]
@sgsokol That bash code is for binary repacking, which is anyway not used for that recipe.

**Comment by dbast** [comment]
@sgsokol Thanks for your contribution!

**Pull request merged**